### PR TITLE
feat(integration): Add integration with Google GenerativeAI SDK

### PR DESCRIPTION
**Pull request by soumik12345**
This PR adds the autopatch integration with [Google GenerativeAI SDK](https://ai.google.dev/api?lang=python).

## Tasks:

- [x] Non async, non streaming
- [x] Non async, streaming
- [x] Async, non streaming
- [x] Async, streaming
- [x] Add tests
- [ ] Add docs

### Non async, non streaming

```python
import os
import google.generativeai as genai
import weave

genai.configure(api_key=os.environ["GOOGLE_API_KEY"])
weave.init(project_name="google_ai_studio-test")
model = genai.GenerativeModel("gemini-1.5-flash")
response = model.generate_content("Write a story about an AI and magic")
```

### Non async, streaming

```python
import os
import google.generativeai as genai
import weave

genai.configure(api_key=os.environ["GOOGLE_API_KEY"])
weave.init(project_name="google_ai_studio-test")
model = genai.GenerativeModel("gemini-1.5-flash")
response = model.generate_content("Write a story about an AI and magic", stream=True)
chunks = [chunk.text for chunk in response]
```

### Async, non streaming

```python
import os
import asyncio
import google.generativeai as genai
import weave

genai.configure(api_key=os.environ["GOOGLE_API_KEY"])
weave.init(project_name="google_ai_studio-test")
model = genai.GenerativeModel("gemini-1.5-flash")

async def async_generate():
    response = await model.generate_content_async(
        "Write a story about an AI and magic"
    )
    return response

response = asyncio.run(async_generate())
```

### Async Streaming

```python
import os
import asyncio
import google.generativeai as genai
import weave

genai.configure(api_key=os.environ["GOOGLE_API_KEY"])
weave.init(project_name="google_ai_studio-test")
model = genai.GenerativeModel("gemini-1.5-flash")

async def get_response():
    async for chunk in await model.generate_content_async(
        "Write a story about an AI and magic", stream=True
    ):
        if chunk.text:
            print(chunk.text)
        print("_" * 80)


response = asyncio.run(get_response())
```

### CODE REVIEW

File: weave/integrations/google_ai_studio/cassettes/google_ai_studio_test/test_content_generation.yaml [hunk 1]
```diff
@@ -0,0 +1,2356 @@
...
+      Accept:
+      - '*/*'
+      Accept-Encoding:
+      - gzip, deflate, zstd
+      Connection:
+      - keep-alive
+      User-Agent:
+      - python-requests/2.32.3
+    method: GET
+    uri: https://raw.githubusercontent.com/BerriAI/litellm/main/model_prices_and_context_window.json
```
Comment (tssweeney) [inline weave/integrations/google_ai_studio/cassettes/google_ai_studio_test/test_content_generation.yaml:14]: please remove this litellm request from this and every other cassette

File: weave/trace_server/migrations/006_seed_costs.up.sql [hunk 2]
```diff
@@ -170,24 +170,24 @@ VALUES
...
-    (generateUUIDv4(), 'default', 'default', 'vertex_ai-language-models', 'gemini-1.5-flash-001', now(), 5e-07, 'USD', 1.5e-06, 'USD', 'system', now()), 
-    (generateUUIDv4(), 'default', 'default', 'vertex_ai-language-models', 'gemini-1.5-flash-preview-0514', now(), 5e-07, 'USD', 1.5e-06, 'USD', 'system', now()), 
-    (generateUUIDv4(), 'default', 'default', 'vertex_ai-language-models', 'gemini-experimental', now(), 0, 'USD', 0, 'USD', 'system', now()), 
-    (generateUUIDv4(), 'default', 'default', 'vertex_ai-vision-models', 'gemini-pro-vision', now(), 2.5e-07, 'USD', 5e-07, 'USD', 'system', now()), 
-    (generateUUIDv4(), 'default', 'default', 'vertex_ai-vision-models', 'gemini-1.0-pro-vision', now(), 2.5e-07, 'USD', 5e-07, 'USD', 'system', now()), 
-    (generateUUIDv4(), 'default', 'default', 'vertex_ai-vision-models', 'gemini-1.0-pro-vision-001', now(), 2.5e-07, 'USD', 5e-07, 'USD', 'system', now()), 
+    (generateUUIDv4(), 'default', 'default', 'vertex_ai-language-models', 'google_ai_studio-pro', now(), 5e-07, 'USD', 1.5e-06, 'USD', 'system', now()),
```
Comment (tssweeney) [inline weave/trace_server/migrations/006_seed_costs.up.sql:173]: This is very very bad. Migrations shall never be edited. Please undo these changes. Note: this data is not used in the production track so not needed anyway

File: weave/integrations/google_ai_studio/cassettes/google_ai_studio_test/test_content_generation.yaml (hunk 1 above)
Comment (soumik12345) [reply weave/integrations/google_ai_studio/cassettes/google_ai_studio_test/test_content_generation.yaml]: Removed the litellm request and responses from the cassettes.
However, it seems like the cassettes are missing the request and responses for the respective ai studio testcases. Any suggestion as to what could be wrong here?

Comment (tssweeney) [reply weave/integrations/google_ai_studio/cassettes/google_ai_studio_test/test_content_generation.yaml]: maybe too much was removed?

Comment (tssweeney) [reply weave/integrations/google_ai_studio/cassettes/google_ai_studio_test/test_content_generation.yaml]: i wonder if your machine cached the results resulting in no calls to record

Comment (soumik12345) [reply weave/integrations/google_ai_studio/cassettes/google_ai_studio_test/test_content_generation.yaml]: I tried on a fresh codespace, still the cassettes are empty other than the litellm ones. This is how I'm generating the cassettes:
```shell
pytest --record-model=rewrite weave/integrations/google_ai_studio/google_ai_studio_test.py::test_content_generation
```

File: requirements.test.txt [hunk 3]
```diff
@@ -43,6 +43,7 @@ chromadb>=0.5.0             # LangChain
 pysqlite3-binary==0.5.3     # LangChain
 cohere>=5.5.8               # Cohere
 groq>=0.9.0                 # Groq
+google-generativeai==0.7.2  # Google Generative AI
```
Comment (andrewtruong) [inline requirements.test.txt:46]: Why pin here?

File: weave/integrations/google_ai_studio/cassettes/google_ai_studio_test/test_content_generation.yaml
Comment (andrewtruong) [inline weave/integrations/google_ai_studio/cassettes/google_ai_studio_test/test_content_generation.yaml:1]: These are blank, but should be filled in.

It should look something like:
https://github.com/wandb/weave/blame/feat/google-ai/weave/integrations/cohere/cassettes/cohere_test/test_cohere.yaml

Comment (soumik12345) [reply weave/integrations/google_ai_studio/cassettes/google_ai_studio_test/test_content_generation.yaml:1]: Yeah, that's the issue, they're not getting filled. This is how I'm generating the cassettes:
```shell
pytest --record-model=rewrite weave/integrations/google_ai_studio/google_ai_studio_test.py::test_content_generation
```

Do you have any suggestions as to what could be wrong here?

Comment (andrewtruong) [reply weave/integrations/google_ai_studio/cassettes/google_ai_studio_test/test_content_generation.yaml:1]: Hey @soumik12345 , some discussion here:
https://weightsandbiases.slack.com/archives/C0785BR1RNZ/p1724159204898399?thread_ts=1723835336.159449&cid=C0785BR1RNZ

File: weave/integrations/openai/cassettes/openai_test/test_openai_as_context_manager_async.yaml
Comment (andrewtruong) [inline weave/integrations/openai/cassettes/openai_test/test_openai_as_context_manager_async.yaml:1]: why did you change this file?

Comment (soumik12345) [reply weave/integrations/openai/cassettes/openai_test/test_openai_as_context_manager_async.yaml:1]: I didn't change this, seems like it got changed somehow when I ran the tests. I'll revert this back.
